src/modules/ztsi/src/lib/Ztsi.cpp

```diff
@@ -205,97 +202,180 @@ bool Ztsi::IsValidConfig(const Ztsi::AgentConfig& config)
     return isValid;
 }
 
-int Ztsi::ReadAgentConfig(Ztsi::AgentConfig& config)
+bool Ztsi::FileExists(const std::string& filePath)
 {
-    int status = 0;
-
-    std::ifstream file(m_agentConfigFile);
-    if (!file.good())
-    {
-        status = ENOENT;
-        return status;
-    }
+    struct stat sb;
+    return (0 == stat(filePath.c_str(), &sb) && S_ISREG(sb.st_mode));
+}
 
-    rapidjson::IStreamWrapper isw(file);
-    rapidjson::Document document;
-    if (document.ParseStream(isw).HasParseError())
-    {
-        OsConfigLogError(ZtsiLog::Get(), "Failed to parse config file: %s", m_agentConfigFile.c_str());
-        status = EINVAL;
-        return status;
-    }
+std::FILE* Ztsi::LockFile(const char* mode)
```

Comment:
>Ztsi::LockFile [](http://example.com/codeflow?start=11&length=14)

OpenAndLockFile? Since this function does more than just locks. Same for UnlockFile

Reply:
Thanks, fixed 👍🏻

Reply:
Thanks!